### DISCUSSION TITLE
Can route be set to be case-insensitive？

### DISCUSSION THREAD

**Discussion by xlg-go** [apache/apisix#5624] [Q&A]
Can route be set to be case-insensitive？

**Comment by tokers** [comment]
Do you mean the match condition for the URI?

**Reply by xlg-go** [reply]
> Do you mean the match condition for the URI?

I hope these two uris can be routed to the same upstream, Only configure one uri instead of two uris.

![image](https://user-images.githubusercontent.com/29967092/143561326-c9a3453d-d954-4402-b267-0c9610fd5e74.png)

**Reply by tokers** [reply]
That's might be a reasonable need.

**Reply by spacewander** [reply]
See https://github.com/apache/apisix/issues/4412

**Reply by xlg-go** [reply]
> That's might be a reasonable need.

@tokers I suggest adding a switch for route matching, open, case-insensitive, close, case-sensitive. Just my personal suggestion. Our company’s business needs such a function, which may be a good need. Hope to adapt~~~
thanks a lot！

**Reply by xlg-go** [reply]
> See #4412

@spacewander Yes, I have see this~ but I have no development experience with Lua. maby, that's might be a reasonable need.
Hope to adapt~~~
thanks a lot！

**Reply by juzhiyuan** [reply]
> but I have no development experience with Lua

Don't worry, you could go through #4412 and other feature implementations and have a try 😄
